### ISSUE TITLE
Add an option to automatically activate the created virtual environment

### ISSUE THREAD

**Issue by Managor** [pypa/virtualenv#2778]
**What's the problem this feature will solve?**

It always feels dumb to have to write another command to activate the venv that you've created.

**Describe the solution you'd like**

I would like to have an option something like `virtualenv -a folder` to automatically source the activate file. This would save me a couple of seconds every time.

**Comment by gaborbernat** [comment]
AFAIK it is not possible for a child process to alter the state of a parent process (your current shell is the parent, virtualenv is the child). So this is a limitation of shells we cannot do much about. You need to call the special shell directive `source`, and only that one is allowed to make this change. Once you call virtualenv we no longer are able to do that.

**Comment by Managor** [comment]
Could this be done if `virtualenv` became the shell? Similar to how tmux creates a new shell on top of the one you called it from.

**Comment by gaborbernat** [comment]
tmux is not a shell, just starts a new shell inside your shell. Which in theory possible, but is a difficult problem to solve with cross OS support. If someone wants to give a stab at it, sure I can review it, but I am not interested in tackling this issue.

**Comment by Managor** [comment]
Yes, that's what I was after. Sorry for the amateurish language. Can you reopen the issue as it's technically possible?

**Comment by gaborbernat** [comment]
Do you plan to work on it?

**Comment by Managor** [comment]
I don't have the expertese. I understand. I'll leave it at that.

**Comment by Managor** [comment]
Oh I'm dumb. I should have just used an alias `alias venv="python -m venv .venv && source .venv/bin/activate"`
Thanks you though. Your comment about the sourcing needing to happen within the same shell forced me to consider my options. I'm leaving this comment here in case anyone is on the same path as I was.
